state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -779,19 +816,35 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			// If the signature verification fails, we want
 			// to retry from the threshold signing stage.
 			else {
-				Self::clean_up_broadcast_storage(broadcast_id);
-				let (retry_broadcast_id, _) = Self::threshold_sign_and_broadcast(
-					api_call,
-					RequestCallbacks::<T, I>::get(broadcast_id),
+				// Only storage to be removed, the other storages corresponding to this broadcast
+				// are either still valid or will get overwritten on re-threshold sign and
+				// broadcast.
+				TransactionOutIdToBroadcastId::<T, I>::remove(broadcast_attempt.transaction_out_id);
+
+				// We update the initiated_at here since as the tx is resigned and broadcast, it is
+				// not possible for it to be successfully broadcasted before this point.
+				let initiated_at = T::ChainTracking::get_block_height();
```

Comment:
What does this mean for witnessing? For example, if one of the previous broadcasts is still pending in the witnesser, does this have any effect?

Reply:
@kylezs maybe you can answer this - will this cause any side effects in the engine?

Reply:
It's possible it does impact witnessing yeah, will have to look a bit deeper to confirm. Relevant code is in `engine/src/witness/common/chunked_chain_source/chunked_by_vault/deposit_addresses.rs` . Otherwise I'll have a look in a bit.

Reply:
Ok, so there might be an issue here - if we're resetting the initiated at this scenario could occur:
1. sign tx with initiated at 20
2. finalised witnessing is at block 22
3. tx gets into block 24 - we have witnessed, since we're slightly behind
4. timeout on SC occurs verification fails
5. block head is at 26, so we resign and set initiated at to block 26
6. Cfe gets to finalised witnessing at block 24, however when it queries storage for what it should witness it filters out the transaction it should because it sees that it's only potentially valid from block 26, so we don't witness it.

so the correct thing to do would be if there already exists a TransactionOutId of this type, we use the lower initiated_at. 

Deleting the `TransactionOutIdToBroadcastId` as is done above here will mean we won't witness it if the transaction is on the external chain in the meantime, which is basically the same issue.

Reply:
also this is a decent chunk of dup'd code

Reply:
So basically we just don't need to do this resetting of the `initiated_at`, right?

Reply:
Yep. But we set it for a new one ofc.

Reply:
The case you are describing above is similar to the one discussed in https://github.com/chainflip-io/chainflip-backend/pull/4262. This can only happen in the unlikely case of the tx witness delay anomaly and the polkadot Runtime version update happening at the same time which themselves own their own, rare events. Moreover, as dan said if in that case we do witness the dot runtime version update, it means the witnessing has been done uptil that point and if the tx hasnt gone through until then, it will fail.

I think it is safe to reset the initiated_at when re signing the tx.